technical-reference/reference-technical-articles/technical-doc-diamond.md

```diff
@@ -0,0 +1,80 @@
+## Overview:
+
+The problem at hand arose when additional features were introduced to the Comptroller contract within the core pool, causing it to surpass the maximum allowable size of 24KB. In response to this challenge and to preemptively address similar issues in the future, a multifaceted solution was implemented.
+
+The solution involves the incorporation of the EIP-2535 pattern into the Comptroller contract, which has been refactored into distinct facets. This restructuring aligns the storage layout with that of the Comptroller proxy, streamlining the contract's organization and enhancing efficiency. Consequently, user interactions now trigger two delegate calls, optimizing the contract's performance and addressing the size constraint effectively.
+## Venus Comptroller - Refactored to EIP 2535
+The previous implementation of Venus used a transparent upgrade proxy pattern with EternalStorage, for the diamond proxy we needed a slightly different implementation of fallback function.
+- Comptroller proxy(Unitroller): https://bscscan.com/address/0xfd36e2c2a6789db23113685031d7f16329158384
+- Current Comptroller implementation: https://bscscan.com/address/0x909dd16b24cef96c7be13065a9a0eaf8a126ffa5
```

Comment:
```suggestion
- Comptroller implementation before enabling the Diamond Proxy: https://bscscan.com/address/0x909dd16b24cef96c7be13065a9a0eaf8a126ffa5
```

I wouldn't use the term "Current" to avoid to have obsolete documentation as soon as we execute the VIP